src/vcpkg-test/files.cpp

```diff
@@ -146,14 +146,126 @@ namespace
     p.make_preferred();
     CHECK(p.native() == expected);
 }
 
+TEST_CASE ("vcpkg Path regular operations", "[filesystem][files]")
+{
+    CHECK(Path().native().empty());
+    Path p("hello");
+    CHECK(p == "hello");
+    CHECK(p.native() == "hello");
+    Path copy_constructed(p);
+    CHECK(copy_constructed == "hello");
+    CHECK(copy_constructed.native() == "hello");
+    Path move_constructed(std::move(p));
+    CHECK(move_constructed == "hello");
+    CHECK(move_constructed.native() == "hello");
+
+    p = "world";
+    Path copy_assigned;
+    copy_assigned = p;
+    CHECK(copy_assigned == "world");
+    CHECK(copy_assigned.native() == "world");
+
+    Path move_assigned;
+    move_assigned = std::move(p);
+    CHECK(move_assigned == "world");
+    CHECK(move_assigned.native() == "world");
+}
+
+TEST_CASE ("vcpkg Path conversions", "[filesystem][files]")
+{
+    StringLiteral sl("some literal");
+    StringView sv = sl;
+    std::string str("some string");
+    std::string moved_from("moved from");
+    const char* ntbs = "some utf-8";
+    CHECK(Path(sv).native() == "some literal");
+    CHECK(Path(str).native() == "some string");
+    CHECK(Path(std::move(moved_from)).native() == "moved from");
+    CHECK(Path(ntbs).native() == "some utf-8");
+    CHECK(Path(str.begin(), str.end()).native() == "some string");
+    CHECK(Path(str.data(), str.size()).native() == "some string");
+
+    Path p("convert from");
+    StringView conv_sv = p;
+    CHECK(conv_sv == "convert from");
+    CHECK(strcmp(p.c_str(), "convert from") == 0);
+}
+
+TEST_CASE ("vcpkg Path generic", "[filesystem][files]")
+{
+    Path p("some/path/with/forward/slashes");
+    CHECK(p.generic_u8string() == StringView("some/path/with/forward/slashes"));
+
+    Path p_dup("some/path/with//////duplicate//////////forward/slashes");
+    CHECK(p_dup.generic_u8string() == StringView("some/path/with//////duplicate//////////forward/slashes"));
+
+    Path bp("some\\path\\/\\/with\\backslashes");
+#if defined(_WIN32)
+    CHECK(bp.generic_u8string() == StringView("some/path////with/backslashes"));
+#else // ^^^ _WIN32 / !_WIN32 vvv
+    CHECK(bp.generic_u8string() == StringView("some\\path\\/\\/with\\backslashes"));
+#endif // _WIN32
+}
+
+static void test_op_slash(StringView base, StringView append, StringView expected) {
+    Path an_lvalue(base);
+    CHECK((an_lvalue / append).native() == expected); // Path operator/(StringView sv) const&;
+    CHECK((Path(base) / append).native() == expected); // Path operator/(StringView sv) &&;
+    an_lvalue /= append; // Path& operator/=(StringView sv);
+    CHECK(an_lvalue.native() == expected);
+}
+
+TEST_CASE ("vcpkg Path::operator/", "[filesystem][files]")
+{
+    test_op_slash("/a/b", "c/d", "/a/b" VCPKG_PREFERED_SEPARATOR "c/d");
+    test_op_slash("a/b", "c/d", "a/b" VCPKG_PREFERED_SEPARATOR "c/d");
+    test_op_slash("/a/b", "/c/d", "/c/d");
+
+#if defined(_WIN32)
+    test_op_slash("C:/a/b", "c/d", "C:/a/b\\c/d");
```

Comment:
Should we duplicate these tests to cover the expected behavior on linux?
e.g.: `"C:/a/b" / "c/d" = "C:/a/b/c/d"`

Reply:
Is `:` valid in file names on Linux?